libcst/codegen/gen_matcher_classes.py

```diff
@@ -511,7 +511,7 @@ def _get_fields(node: Type[cst.CSTNode]) -> Generator[Field, None, None]:
 
     generated_code.append("")
     generated_code.append("")
-    generated_code.append("@dataclass(frozen=True)")
+    generated_code.append("@dataclass(frozen=True, eq=False, unsafe_hash=False)")
```

Comment:
The default value of `unsafe_hash` is False, we can just skip it.

Reply:
No, it is true when `frozen=True`, hence why there was a hash generated for these classes that cause the problem this diff is addressing.

Reply:
>>> If eq and frozen are both true, by default dataclass() will generate a __hash__() method for you. If eq is true and frozen is false, __hash__() will be set to None, marking it unhashable (which it is, since it is mutable). If eq is false, __hash__() will be left untouched meaning the __hash__() method of the superclass will be used (if the superclass is object, this means it will fall back to id-based hashing).
https://docs.python.org/3/library/dataclasses.html

Based on CPython document, when `frozen=True and eq=False`, the __hash__ will left untouched and fall back to id-based when superclass is object, which seems apply to our case?

Reply:
I verified it and the document is correct that when frozen and eq are True, setting unsafe_hash=False or not won't change the behavior of the hash function and it's the identity.

Reply:
Without these changes, we can't hash list and the provided unit test fails. Perhaps one of these can be dropped, but it seems foolish to rely on defaults when we want to explicitly dictate behavior. Especially given how unintelligible dataclass docs are with respect to what values are defaulted given other values.